src/core/thread/mle.cpp

```diff
@@ -380,6 +382,9 @@ otError Mle::Restore(void)
     Get<KeyManager>().SetMacFrameCounter(networkInfo.GetMacFrameCounter());
     mDeviceMode.Set(networkInfo.GetDeviceMode());
 
+    // force re-attach when version mismatch.
+    VerifyOrExit(networkInfo.GetVersion() == kThreadVersion);
```

Comment:
Return error properly ?

Reply:
@bukepo

Reply:
I don't think this is error.

Reply:
If the question is "Will a node restore successfully when version mismatch?". I think the answer should be "No." So, there should be an error. Thoughts?

Reply:
Should we add a test verifying this?

Reply:
@edmont agree to add test on this. 
//cc// @bukepo

Reply:
After rethink over this, the test should be added after this PR gets merged?
As the test would require both 1.1 and 1.2 build, and it is supported in the script in this PR
Thoughts?
@bukepo @edmont

Reply:
@librasungirl yes, agree to add the test later on.